py/dml/template.py

```diff
@@ -123,15 +128,28 @@ def defined_symbols(self):
                     assert sub.kind == 'error'
             for (_, name, _, specs) in composite:
                 symbols[name] = ('subobj', specs.site)
+        self._defined_symbols = symbols
```

Comment:
Memoization is cause of the use in `mkTraitTemplateQualifiedMethodRef`. I suppose `mkTraitTemplateQualifiedMethodRef` could leverage the new `get_potential_method_impl_details` instead?

Reply:
why don't you use `@functools.cache` to memoize?

Reply:
cause i didn't know it existed

Reply:
`@functools.cache` now used. `mkTraitTemplateQualifiedMethodRef` now uses `get_potential_method_impl_details()`, so the memoization for `defined_symbols()` has been completely removed.